src/plugins/wb-data-scrub/wb-data-scrub.js

```diff
@@ -0,0 +1,207 @@
+						if ( !form.classList.contains( "wb-postback" ) ) {
+							event.currentTarget.submit();
+						} else {
+							$( form ).trigger( "wb-postback.submit", { event } );
+						}
+					}
+				} );
+
+				// This is to prevent this event listener to be added for
+				// every field with the attribute "data-rule-wbscrub"
+				form.isPIIinited = true;
+			}
+
+			wb.ready( $( elm ), componentName );
+		}
+	},
+
+	/*
+	* Log all PII positive fields inside the form's "PIIFields" property
+	* @param form: a reference to the form containing PII fields
+	*/
+	checkFormValues = function( form ) {
+		let fieldsToScrub = form.querySelectorAll( selector );
+
+		form.PIIFields = [];
+
+		// identify form elements that were assigned to be scrubbed
+		fieldsToScrub.forEach( ( field ) => {
+
+			// If the field contains PII add field to list
+			if ( wb.findPotentialPII( field.value, false ) ) {
+				let currFieldValue = field.value,
+					scrubbedFieldValue = wb.findPotentialPII( field.value, true, { replaceWith: "********" } ),
+					fieldLabel = form.querySelector( "[for=" + field.id + "] > span.field-name" ),
+					fieldLabelText = fieldLabel ? fieldLabel.innerText : form.querySelector( "[for=" + field.id + "]" ).innerText;
+
+				form.PIIFields.push( {
+					id: field.id,
+					currVal: currFieldValue,
+					scrubVal: scrubbedFieldValue,
+					label: fieldLabelText
+				} );
+			}
+		} );
+
+		if ( form.PIIFields.length === 1 ) {
+			document.getElementById( form.PIIFields[ 0 ].id ).focus();
+		}
+
+		// If PII is found, block Postback form submission
+		form.PIIFields.length > 0 ? form.setAttribute( attrBlocked, "true" ) : form.setAttribute( attrBlocked, "false" );
+	},
+
+	/*
+	* Scrub all PII positive fields
+	* @param form: a reference to the form containing PII fields
+	*/
+	scrubFormValues = function( form ) {
+
+		// Scrub the value of each PII positive fields
+		form.PIIFields.forEach( ( field ) => {
+			let fieldElm = document.getElementById( field.id );
+
+			fieldElm.value = field.scrubVal;
+		} );
+
+		// Clear PII fields as their value has been replaced by the scrubbed value
+		form.PIIFields = [];
+
+		// Unblock form submission as there are no longer PII concerns
+		form.removeAttribute( attrBlocked );
+	},
+
+	/*
+	* Generate the modal UI
+	* @param form: a reference to the form containing PII fields
+	*/
+	generateModal = function( form ) {
+		let piiModalFields = "",
+			piiModal = document.createElement( "section" );
+
+		// Destroy modal if present
+		if ( document.getElementById( piiModalID ) ) {
+			document.getElementById( piiModalID ).remove();
+		}
+
+		// Generate PII fields list
+		form.PIIFields.forEach( ( field ) => {
+			piiModalFields += "<dt>" + field.label + "</dt><dd><code>" + field.scrubVal + "</code></dd>";
+		} );
+
+		piiModal.id = piiModalID;
+		piiModal.className = "modal-dialog modal-content overlay-def";
+		piiModal.innerHTML = `<header class="modal-header">
```

Comment:
Please check that all inner variable are safe. Or parse this through DOM purify at least.